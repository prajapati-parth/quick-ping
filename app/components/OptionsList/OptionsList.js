import React, {Component} from 'react'
import {
	Text,
	View,
    ListView
} from 'react-native'
import CompStyles from './comp_styles'
import Options from '../Options'

export default class OptionsList extends Component {
    constructor(props) {
        super(props)
        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        this.state = {
            dataSource: ds.cloneWithRows(this.props.selectOptions),
        };
    }

    render() {
        return (
            <View>
                <ListView
                    dataSource={this.state.dataSource}
                    renderRow={
                        (rowData) => <Options
                                        name={rowData.name}
                                    />
                    }
                />
            </View>
        )
    }
}