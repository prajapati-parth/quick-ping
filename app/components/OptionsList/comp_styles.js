import {StyleSheet} from 'react-native'
import GlobStyles from '../../config/styles'

const CompStyles = StyleSheet.create({
    optionContainer: {
        flex: 1,
        paddingLeft: GlobStyles.boundaryPadding,
        paddingRight: GlobStyles.boundaryPadding,
        marginBottom: GlobStyles.boundaryPadding/2
    }
})

export default CompStyles