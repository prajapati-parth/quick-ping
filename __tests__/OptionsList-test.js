import 'react-native'
import React from 'react'
import OptionsList from '../app/components/OptionsList'
import SelectOptions from '../app/config/options'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
    const tree = renderer.create(
        <OptionsList
            selectOptions={SelectOptions}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
})