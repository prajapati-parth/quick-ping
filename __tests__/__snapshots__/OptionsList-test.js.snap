// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 4,
      }
    }
    renderRow={[Function]}
    renderScrollComponent={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Empty water(302)
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Empty water(202)
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Empty coffee(302)
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Empty tea(302)
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Empty coffee(202)
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Empty tea(202)
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Need cleaning(302)
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Need cleaning(202)
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
